code/modules/crafting/table.dm

```diff
@@ -164,11 +175,46 @@
 		dat += "Crafting in progress...</div>"
 	else
 		for(var/datum/table_recipe/R in table_recipes)
+			var/name_text = ""
+			var/req_text = ""
+			var/tool_text = ""
+			var/catalist_text = ""
 			if(check_contents(R))
-				dat += "<A href='?src=\ref[src];make=\ref[R]'>[R.name]</A><BR>"
+				name_text ="<A href='?src=\ref[src];make=\ref[R]'>[R.name]</A>"
+
+			else if(partial_check_contents(R))
+				name_text = "<span class='linkOff'>[R.name]</span>"
+
+			if(name_text)
+				for(var/A in R.reqs)
+					if(ispath(A, /obj))
+						var/obj/O = new A
+						req_text += " [R.reqs[A]] [O.name]"
+						qdel(O)
+					if(ispath(A, /datum/reagent))
+						var/datum/reagent/RE = new A
+						req_text += " [R.reqs[A]] [RE.name]"
+						qdel(RE)
+
+				if(R.chem_catalists.len)
+					catalist_text += ", Catalists:"
```

Comment:
The spelling is Catalyst, unless you're making a pun since this is a list of objects.